packages/next/server/next-server.ts

```diff
@@ -126,15 +126,19 @@ export default class NextNodeServer extends BaseServer {
       )
     }
 
-    if (!this.renderOpts.dev) {
+    if (!options.dev) {
       // pre-warm _document and _app as these will be
       // needed for most requests
-      loadComponents(this.distDir, '/_document', this._isLikeServerless).catch(
-        () => {}
-      )
-      loadComponents(this.distDir, '/_app', this._isLikeServerless).catch(
-        () => {}
-      )
+      try {
+        loadComponents(
+          this.distDir,
+          '/_document',
+          this._isLikeServerless
+        ).catch(() => {})
+        loadComponents(this.distDir, '/_app', this._isLikeServerless).catch(
+          () => {}
+        )
+      } catch (_) {}
```

Comment:
The try catch won't help here, since it's a promise rejection...
`loadComponents` will not throw, but returns a rejected promise.

Reply:
```
    const [DocumentMod, AppMod, ComponentMod, AppServerMod] = await Promise.all([
        Promise.resolve().then(() => requirePage('/_document', distDir, serverless)),
        Promise.resolve().then(() => requirePage('/_app', distDir, serverless)),
        Promise.resolve().then(() => requirePage(pathname, distDir, serverless)),
        serverComponents ? requirePage('/_app.server', distDir, serverless) : null,
    ]);
```

in `loadComponents` helped for me